src/gsy_e/gsy_e_core/simulation.py

```diff
@@ -357,7 +358,9 @@ def update_and_send_results(self, current_state: dict, progress_info: Simulation
         """
         assert self._endpoint_buffer is not None
 
-        if ConstSettings.MASettings.MARKET_TYPE == SpotMarketTypeEnum.COEFFICIENTS:
+        if ConstSettings.MASettings.MARKET_TYPE == SpotMarketTypeEnum.COEFFICIENTS.value:
+            if not self._should_send_results_to_broker:
+                return
```

Comment:
I will omit this because I have significantly refactored the simulation results for the coefficients, and remove this if block completely as part of it (I instead used factory classes to either construct an instance of SimulationResultsManager or CoefficientSImulationResultsManager to avoid the need for these if blocks). Therefore if you can be patient till the followup PR would be great!